File: Src2/Icm.ContextConsole/DependencyResolver/StandardDependencyResolver.vb
Public Class StandardDependencyResolver
    Implements IDependencyResolver


    Private store As New Dictionary(Of Type, Object)

    Public Overridable Function GetService(service As Type) As Object Implements IDependencyResolver.GetService
        Dim result As Object = Nothing
        If Not store.TryGetValue(service, result) Then
            result = CreateService(service)
            store.Add(service, result)
        End If
        Return result
    End Function

    Protected Overridable Function CreateService(service As Type) As Object
        Return Activator.CreateInstance(service)
    End Function

    Public Sub ClearRequestScope() Implements IDependencyResolver.ClearRequestScope
        store.Clear()
    End Sub
End Class

File: Src/Icm.ContextConsole/DependencyResolver/StandardDependencyResolver.cs
using System;
using System.Collections.Generic;

public class StandardDependencyResolver : IDependencyResolver
{
	private Dictionary<Type, object> store = new Dictionary<Type, object>();
	public virtual object GetService(Type service)
	{
		object result = null;
		if (!store.TryGetValue(service, out result)) {
			result = CreateService(service);
			store.Add(service, result);
		}
		return result;
	}

	protected virtual object CreateService(Type service)
	{
		return Activator.CreateInstance(service);
	}

	public void ClearRequestScope()
	{
		store.Clear();
	}
}

//=======================================================
//Service provided by Telerik (www.telerik.com)
//Conversion powered by NRefactory.
//Twitter: @telerik
//Facebook: facebook.com/telerik
//=======================================================
